scRNA-seq/04-dimension_reduction_scRNA.Rmd

```diff
@@ -31,225 +44,255 @@ library(magrittr)
-qc_pass <- (tm_sce$detected > 500) & (tm_sce$subsets_mito_percent < 20)
-tm_sce_filtered <- tm_sce[, qc_pass]
+```{r}
+filtered_sce <- filtered_sce[, filtered_sce$subsets_mito_percent < 20]
 ```
 
+
 We can also filter by features (genes in our case) using `scater::addPerFeatureQC()` which will computer the number of samples where each gene is detected and the mean count across all genes. 
-We can then use those data (stored in `rowData`) to filter by row to only the genes that are detected in at least 5 cells, and with a mean count > 0.1.
+We can then use those data (stored in `rowData`) to filter by row to only the genes that are detected in at least 5% of cells, and with a mean count > 0.1.
 
 ```{r gene_qc}
-tm_sce_filtered <- scater::addPerFeatureQC(tm_sce_filtered)
-detected <- rowData(tm_sce_filtered)$detected > 5
-expressed <- rowData(tm_sce_filtered)$mean > 0.1
+filtered_sce <- scater::addPerFeatureQC(filtered_sce)
+detected <- rowData(filtered_sce)$detected > 5
+expressed <- rowData(filtered_sce)$mean > 0.1
 
 # filter the genes (rows) this time
-tm_sce_filtered <- tm_sce_filtered[detected & expressed, ]
+filtered_sce <- filtered_sce[detected & expressed, ]
 ```
 
+```{r}
+dim(filtered_sce)
+```
+
+
 ### Normalize
 
-Now we will perform the same normalization steps we did with the Smart-seq2 data, using `scran::computeSumFactors()` and `scater::logNormCounts()`
+Now we will perform the same normalization steps we did with a previous dataset, using `scran::computeSumFactors()` and `scater::logNormCounts()`
 
 ```{r normalize}
 # Cluster similar cells
-qclust <- scran::quickCluster(tm_sce_filtered)
+qclust <- scran::quickCluster(filtered_sce)
 
 # Compute sum factors for each cell cluster grouping.  
-tm_sce_filtered <- scran::computeSumFactors(tm_sce_filtered, clusters = qclust)
+filtered_sce <- scran::computeSumFactors(filtered_sce, clusters = qclust)
 
 # Normalize and log transform. 
-tm_sce_filtered <- scater::logNormCounts(tm_sce_filtered)
+normalized_sce <- scater::logNormCounts(filtered_sce)
 ```
 
+At this point, we have a few different versions of our `SingleCellExperiment` object.
+The original (mostly) unfiltered version is in `hodgkins_sce`, the filtered version in `filtered_sce`, and the normalized version in `normalized_sce`.
+We can clean those up a bit to save memory, keeping only the latest `normalized_sce` version, which now has two `assay`s: 
+`counts` with the raw data and `logcounts` with the normalized and transformed data.
+
+```{r clean_up, live = TRUE}
+assayNames(normalized_sce)
+rm(hodgkins_sce, filtered_sce)
+```
+
+
 
 ![**Status of the current dataset**](diagrams/tag-based_3.png)
 
-## Principal Components Analysis
+
+
+## Dimensionality reduction and display
+
+### Principal Components Analysis
 
 The first dimensionality reduction algorithm we will use is PCA, which we have already used!
 The idea of PCA is that it finds the axes which explain the greatest amount of variance in the data.
```

Comment:
That will not be true during this training as far as I know, so would port over some of the text in: https://github.com/AlexsLemonade/training-modules/blob/0388bf1c7c04fbbc5dd3ff64f8eae98b2472826c/RNA-seq/03-gastric_cancer_exploratory.Rmd#L112